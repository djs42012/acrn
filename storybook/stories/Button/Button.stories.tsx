import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../CenterView';
import Button from '../../../src/components/Button/Button';

storiesOf('Button', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('Demo', () => (
    <Button
      text="Button"
      onPressIn={() => console.log('onPressIn')}
      onPress={() => console.log('onPress')}
      onPressOut={() => console.log('onPressOut')}
      onLongPress={() => console.log('onLongPress')}
    />
  ));
