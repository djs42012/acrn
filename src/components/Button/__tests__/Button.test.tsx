import React from 'react';
import Button from '../Button';
import { render, fireEvent } from '@testing-library/react-native';

describe('Accessible Button', () => {
  let button: any;
  const onPress = jest.fn();
  const props = {
    text: 'New Button',
  };
  beforeEach(() => {
    button = render(<Button {...props} onPress={onPress} />);
    expect(button).toMatchSnapshot();
  });
  test('renders input text', () => {
    expect(button.getByText('New Button')).toBeTruthy();
  });
  test('button is pressable', () => {
    fireEvent.press(button.getByText('New Button'));
    expect(onPress).toHaveBeenCalled();
  });
  test('accessibility role is button', () => {
    expect(button.getByRole('button')).toBeTruthy();
  });
});
