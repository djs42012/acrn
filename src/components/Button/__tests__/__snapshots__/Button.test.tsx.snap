// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accessible Button accessibility role is button 1`] = `
<View
  accessibilityLabel="New Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "purple",
      "borderRadius": 20,
      "elevation": 4,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
  text="New Button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    New Button
  </Text>
</View>
`;

exports[`Accessible Button button is pressable 1`] = `
<View
  accessibilityLabel="New Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "purple",
      "borderRadius": 20,
      "elevation": 4,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
  text="New Button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    New Button
  </Text>
</View>
`;

exports[`Accessible Button renders input text 1`] = `
<View
  accessibilityLabel="New Button"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "purple",
      "borderRadius": 20,
      "elevation": 4,
      "justifyContent": "center",
      "marginVertical": 5,
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
  text="New Button"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    New Button
  </Text>
</View>
`;
